orange-tree-lunch.rb

```diff
@@ -0,0 +1,74 @@
+# Orange Tree Group Project - Loraine, Michelle, and Corinne!
+
+class OrangeTree
+  attr_reader :height, :age, :life
+
+  def initialize()
+    @height = 1 # trees are born at the height of 1 unit something
+    @age = 0
+    @orange_count = 0
+    @life = true # life will return true as long as the tree is alive
+  end
+
+  def count_the_oranges
+    return @orange_count
+  end
+
+  # Picks an orange off the tree (only if you have oranges on your tree!)
+  def pick_an_orange
+    if @orange_count > 0
+      @orange_count -= 1
+      puts "Yumm! How delicious!"
+    elsif @orange_count == 0
+      puts "There are no oranges left to pick. Sorry!"
+    end
+  end
+
+  # If the tree is alive, we update the height, age, and orange count
+  def one_year_passes
+    if @life == true
+      @height += 1
+      @age += 1
+      if @age >= 3 && @age <= 5 # orange count is different depending on the age
+        @orange_count = 20
+      elsif @age >= 5 && @age < 7
+        @orange_count = 30
+      elsif @age == 7 # orange tree dies at age 7, orange count is zero upon death
+        @life = false
+        @orange_count = 0
+        puts "The tree has died."
+      end
+    elsif @life == false # tree does not age/grow/grow oranges once it dies
+      puts "Plant a tree. This one's dead. :\("
+    end
```

Comment:
This method is still fairly short and fine, but when you find yourself writing many if/else conditions inside each other, it might be a good flag to either simplify, or break out into multiple methods for clarity. I still think this method is very easy to follow-- just something to watch out for in the future!

I do think, if you wanted to, you could simplify your conditions by re-ordering them. If you start with if @age == 7, then you will not need to write a statement checking that it's less than 7 afterwards, and then same for @age >= 5. Let me know if you have any questions about this!